### PR TITLE
Updated versions of Clojure and lein-ring plugin, resolving crash

### PR DESCRIPTION
**Pull request by jcorrado**
* Updated to Clojure `1.10.0`, same as yetibot.
* Updated lein-ring plugin to `0.12.5`.  The version in yetibot, `0.12.4`, did not resolve the crash on launch.

I didn't investigate updating other deps.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,4 @@ pom.xml.asc
 .lein-failures
 /resources/public/
 node_modules/
+/resources/templates/.sass-cache
```
Comment (devth) [inline .gitignore]: I think I've been checking in the .sass-cache so far. Think we should keep it out?

**Comment by jcorrado** [comment]
For posterity, here are details of the crash on launch:
```
~/projects/yetibot/yetibot.github.io source                                           
❯ PORT=4040 lein ring server                                                    
                                                     
clojure.lang.Compiler$CompilerException: Syntax error macroexpanding clojure.core/fn at (clojure/core/unify.clj:83:18).
#:clojure.error{:phase :macro-syntax-check, :line 83, :column 18, :source "clojure/core/unify.clj", :symbol clojure.core/fn}
 at clojure.lang.Compiler.checkSpecs (Compiler.java:6971)                    
    clojure.lang.Compiler.macroexpand1 (Compiler.java:6987)            
    clojure.lang.Compiler.analyzeSeq (Compiler.java:7092)
    clojure.lang.Compiler.analyze (Compiler.java:6789)
```

A little further down the stack trace:
```
Caused by: clojure.lang.ExceptionInfo: Call to clojure.core/fn did not conform to spec.
#:clojure.spec.alpha{:problems ({:path [:fn-tail :arity-1 :params], :pred clojure.core/vector?, :val clojure.core.unify/var-unify, :via [:clojure.core.specs.alpha/params+body :clojure.core.specs.alpha/param-li
st :clojure.core.specs.alpha/param-list], :in [0]} {:path [:fn-tail :arity-n], :pred (clojure.core/fn [%] (clojure.core/or (clojure.core/nil? %) (clojure.core/sequential? %))), :val clojure.core.unify/var-unif
y, :via [:clojure.core.specs.alpha/params+body :clojure.core.specs.alpha/params+body], :in [0]}), :spec #object[clojure.spec.alpha$regex_spec_impl$reify__2509 0x6d3a388c "clojure.spec.alpha$regex_spec_impl$rei
fy__2509@6d3a388c"], :value (clojure.core.unify/var-unify [G__800 G__801 G__802 G__803] (clojure.core/if-let [vb__793__auto__ (G__803 G__801)] (clojure.core.unify/garner-unifiers G__800 vb__793__auto__ G__802
G__803) (clojure.core/if-let [vexpr__794__auto__ (clojure.core/and (G__800 G__802) (G__803 G__802))] (clojure.core.unify/garner-unifiers G__800 G__801 vexpr__794__auto__ G__803) (if (clojure.core.unify/occurs?
 G__800 G__801 G__802 G__803) (throw (java.lang.IllegalStateException. (clojure.core/str "Cycle found in the path " G__802))) (clojure.core.unify/bind-phase G__803 G__801 G__802))))), :args (clojure.core.unify
/var-unify [G__800 G__801 G__802 G__803] (clojure.core/if-let [vb__793__auto__ (G__803 G__801)] (clojure.core.unify/garner-unifiers G__800 vb__793__auto__ G__802 G__803) (clojure.core/if-let [vexpr__794__auto_
_ (clojure.core/and (G__800 G__802) (G__803 G__802))] (clojure.core.unify/garner-unifiers G__800 G__801 vexpr__794__auto__ G__803) (if (clojure.core.unify/occurs? G__800 G__801 G__802 G__803) (throw (java.lang
.IllegalStateException. (clojure.core/str "Cycle found in the path " G__802))) (clojure.core.unify/bind-phase G__803 G__801 G__802)))))}
 at clojure.spec.alpha$macroexpand_check.invokeStatic (alpha.clj:705)
    clojure.spec.alpha$macroexpand_check.invoke (alpha.clj:697)          
    clojure.lang.AFn.applyToHelper (AFn.java:156)      
```

File: .gitignore (hunk 1 above)
Comment (jcorrado) [reply .gitignore]: Actually, I should have checked to see that you were committing this.  I just assumed it was newly created on my system.  Habit of excluding things named `*cache*` I guess :)

I don't know anything about SASS, so if it's safe and common to keep that, I'll back that exclude out.

Comment (devth) [reply .gitignore]: Doesn't hurt - let's just keep it for now. This is kind of a cobbled-togther asset pipeline. (I added JS bundling separately using Webpack last week).

**Pull request merged**